docs/api/upgrade-assistant/reindexing.asciidoc

```diff
@@ -0,0 +1,151 @@
+[[upgrade-assistant-api-reindexing]]
+=== Reindex API
+
+experimental[This API is *experimental* and may be changed or removed completely in a future release. The underlying Spaces concepts are stable, but the APIs for managing Spaces are currently experimental.]
+
+When checking the upgrade status, some indices will have the `reindex` paramaeter set to true. You can use this Reindexing API to reindex these indices.
+
+==== Start or resume a reindex
+
+===== Request
+
+To start a new reindex or resume a paused reindex, submit a POST request to the `/api/upgrade_assistant/reindex/<indexName>` endpoint:
+
+Note: You cannot access this endpoint via the Console in Kibana.
+
+[source,js]
+--------------------------------------------------
+POST /api/upgrade_assistant/reindex/myIndex
+--------------------------------------------------
+// KIBANA
+
+===== Response
+
+A successful call returns a response code of `200` and a response body
+containing a JSON structure similar to the following example:
+
+[source,js]
+--------------------------------------------------
+{
+  "indexName": ".ml-state",
+  "newIndexName": ".reindexed-v7-ml-state",
+  "status": 0,
+  "lastCompletedStep": 0,
+  "reindexTaskId": null,
+  "reindexTaskPercComplete": null,
+  "errorMessage": null
+}
+--------------------------------------------------
+
+See the next section below for an explanation of each field.
+
+==== Check the status of a reindex
+
+===== Request
+
+Once a reindex has been started, you can check the status of the reindex operation by submitting a GET request to the `/api/upgrade_assistant/reindex/<indexName>` endpoint:
+
+Note: You cannot access this endpoint via the Console in Kibana.
+
+[source,js]
+--------------------------------------------------
+GET /api/upgrade_assistant/reindex/myIndex
+--------------------------------------------------
+// KIBANA
+
+===== Response
+
+A successful call returns a response code of `200` and a response body
+containing a JSON structure similar to the following example:
+
+[source,js]
+--------------------------------------------------
+{
+  "reindexOp": {
+    "indexName": ".ml-state",
+    "newIndexName": ".reindexed-v7-ml-state", <1>
+    "status": 0, <2>
+    "lastCompletedStep": 40, <3>
+    "reindexTaskId": "QprwvTMzRQ2MLWOW22oQ4Q:11819", <4>
+    "reindexTaskPercComplete": 0.3, <5>
+    "errorMessage": null <6>
+  },
+  "warnings": [], <7>
+  "hasRequiredPrivileges": true <8>
+}
+--------------------------------------------------
+
+<1> Name of the new index that is being created.
+<2> Current status of the reindex. See the status code table below for details.
```

Comment:
Would be nice to make mentions of "status code table" and "step code table" and "warning code table" link to the appropriate sections.  If so, you can remove the word "below"

Reply:
Good idea, done!